integration/testutil.go

```diff
@@ -718,3 +719,42 @@ func teardown(cleanups []func() error) error {
 	}
 	return nil
 }
+
+// middleSizeLayerInfo finds a layer not the smallest or largest (if possible), returns index, size, and layer count
+func middleSizeLayerInfo(t *testing.T, sh *shell.Shell, image imageInfo) (int, int64, int) {
+	rebootContainerd(t, sh, getContainerdConfigToml(t, false), getSnapshotterConfigToml(t, false))
```

Comment:
why is `rebootContainerd` needed here?

Reply:
`rebootContainerd` is also secretly "start containerd and/or soci-snapshotter-grpc when they aren't already running". Refactoring to split out start/restart/stop for the two services would be useful in a lot of places.

Reply:
Opened #401 based on this.

Reply:
I think `middleSizeLayerInfo`, being a test helper func, shouldn't reboot containerd (and override configs with default) and can assume there is containerd/soci running (similar to `buildIndex` assuming it).

This also might cause unexpected behavior change (e.g. override `additionalConfigs` set up by a test itself, when a test just want to get a value from the helper). Whether or not reboot should be set up by the test itself.

As an exmaple, `TestLazyPullWithSparseIndex` doesn't need to reboot after calling the helper, since it only requires the value. Whereas `TestRpullImageWithMinLayerSize` need to call `rebootContainerd` after `middleSizeLayerInfo`.